voting/tests/common/scenarios.rs

```diff
@@ -0,0 +1,161 @@
+    }
+
+    fn has(&self, key: &[u8]) -> bool {
+        self.map.get(&blake256(key)).is_some()
+    }
+
+    fn create_checkpoint(&mut self) {
+        unimplemented!()
+    }
+
+    fn discard_checkpoint(&mut self) {
+        unimplemented!()
+    }
+
+    fn revert_to_the_checkpoint(&mut self) {
+        unimplemented!()
+    }
+}
+
+fn tx_create_general_meeting(
+    public: &Public,
+    private: &Private,
+    number_of_agendas: u32,
+    end_time: u64,
+    tallying_time: u64,
+) -> Transaction {
+    let tx = vote::general_meeting::TxCreateGeneralMeeting {
+        number_of_agendas,
+        voting_end_time: TimeStamp {
+            time: end_time,
+        },
+        tallying_time: TimeStamp {
+            time: tallying_time,
+        },
+    };
+    let tx = UserTransaction {
+        network_id: Default::default(),
+        action: tx,
+    };
+    let tx_hash = tx.hash();
+    let tx = SignedTransaction {
+        signature: ckey::sign(&tx_hash.as_bytes(), private),
+        signer_public: *public,
+        tx,
+    };
+    Transaction::new("CreateGeneralMeeting".to_owned(), serde_cbor::to_vec(&tx).unwrap())
+}
+
+pub fn test_create_meeting(ctx: &RwLock<Context>) {
+    for stateful in ctx.write().statefuls.values_mut() {
+        stateful.set_storage(ServiceRef::create_export(Box::new(MockDb::default()) as Box<dyn SubStorageAccess>))
+    }
+
+    let admin: Ed25519KeyPair = Random.generate().unwrap();
+    let fake_admin: Ed25519KeyPair = Random.generate().unwrap();
+
+    let admin_key = admin.public();
+
+    ctx.write()
+        .init_genesises
+        .get_mut("general_meeting")
+        .unwrap()
+        .init_genesis(&serde_cbor::to_vec(&admin_key).unwrap());
+
+    let create_meeting = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 12, 18);
+    let create_meeting_invalid = tx_create_general_meeting(fake_admin.public(), fake_admin.private(), 2, 12, 18);
+
+    let create_meeting_invalid_end_time = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 19, 18);
+    let create_meeting_end_passed = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 0, 18);
+    let create_meeting_tallying_passed = tx_create_general_meeting(&admin.public(), &admin.private(), 2, 0, 0);
+
+    let block_hash: BlockHash = BlockHash::default();
+    let author: Public = Public::default();
+    let mut validators: Vec<Public> = Vec::new();
+    validators.push(Public::default());
+    let extera_data: Bytes = Bytes::default();
+
+    let header = Header::new(block_hash, 1, 0, author, validators, extera_data);
+    ctx.write().tx_owners.get_mut("general_meeting").unwrap().block_opened(&header).unwrap();
+    ctx.write().tx_owners.get_mut("general_meeting").unwrap().execute_transaction(&create_meeting).unwrap();
+    assert!(ctx
+        .write()
+        .tx_owners
+        .get_mut("general_meeting")
+        .unwrap()
+        .execute_transaction(&create_meeting_invalid)
+        .is_err());
```

Comment:
There are too many dots. Please use variables for them. There is a term related these dots: https://en.wikipedia.org/wiki/Law_of_Demeter

Reply:
Please check which error is returned.